transport/graphsync/hooks.go

```diff
@@ -13,44 +16,24 @@ import (
 
 // gsOutgoingRequestHook is called when a graphsync request is made
 func (t *Transport) gsOutgoingRequestHook(p peer.ID, request graphsync.RequestData, hookActions graphsync.OutgoingRequestHookActions) {
-	message, _ := extension.GetTransferData(request, t.supportedExtensions)
 
-	// extension not found; probably not our request.
-	if message == nil {
+	chid, ok := t.requestIDToChannelID.load(request.ID())
```

Comment:
can you add (back) a comment here to say why we might not get one for this `request.ID()`? is it simply going to be a spoof (or mistake) from the peer?

Reply:
fair enough